includes/utilities.inc

```diff
@@ -289,6 +290,16 @@ function islandora_solr_prepare_solr_results($solr_results) {
         $value = $highlighting[$pid][$field];
       }
 
+      // Truncate value lengths before linking, avoids destroying link tags.
+      if (array_key_exists($field, $truncate_length)) {
+        array_walk($value, function (&$val_val, $val_key) use ($field, $truncate_length) {
+          $val_val = strlen($val_val) > intval($truncate_length[$field]['maximum_length']) ?
+            substr($val_val, 0, intval($truncate_length[$field]['maximum_length'])) .
+            $truncate_length[$field]['truncation_suffix'] :
+            $val_val;
+        });
```

Comment:
Should possibly use [`truncate_utf8()`](https://api.drupal.org/api/drupal/includes!unicode.inc/function/truncate_utf8/7), instead of reimplementing... Using it directly, you would lose control of the suffix, but one could compare the lengths of the strings to determine whether or not to tack it on instead.